src/openrct2/object/WaterObject.cpp

```diff
@@ -55,3 +59,85 @@ void WaterObject::DrawPreview(rct_drawpixelinfo * dpi, sint32 width, sint32 heig
     sint32 y = height / 2;
     gfx_draw_string_centred(dpi, STR_WINDOW_NO_IMAGE, x, y, COLOUR_BLACK, nullptr);
 }
+
+void WaterObject::ReadJson(IReadObjectContext * context, const json_t * root)
+{
+    auto properties = json_object_get(root, "properties");
+    _legacyType.flags = ObjectJsonHelpers::GetFlags<uint16>(properties, {
+        { "allowDucks", WATER_FLAGS_ALLOW_DUCKS }});
+
+    ObjectJsonHelpers::LoadStrings(root, GetStringTable());
+
+    // Images which are actually palette data
+    static const char * paletteNames[] =
+    {
+        "general",
+        "waves-0",
+        "waves-1",
+        "waves-2",
+        "sparkles-0",
+        "sparkles-1",
+        "sparkles-2"
+    };
+    for (auto paletteName : paletteNames)
+    {
+        auto jPalettes = json_object_get(properties, "palettes");
+        if (jPalettes != nullptr)
+        {
+            auto jPalette = json_object_get(jPalettes, paletteName);
+            if (jPalette != nullptr)
+            {
+                ReadJsonPalette(jPalette);
+            }
+        }
+    }
+}
+
+void WaterObject::ReadJsonPalette(const json_t * jPalette)
+{
+    auto paletteStartIndex = json_integer_value(json_object_get(jPalette, "index"));
+    auto jColours = json_object_get(jPalette, "colours");
+    auto numColours = json_array_size(jColours);
+
+    auto data = std::make_unique<uint8[]>(numColours * 3);
```

Comment:
`g1_calculate_data_size`?

Reply:
Nah, I don't think that makes sense. The following code is producing data that is assumed to be `numColours * 3`. It would only make sense to call `g1_calculate_data_size` as an assert - but even then you just asserting a constant.